ios/Classes/NativeBridge/ModuleExtensions/OneginiModuleSwift+Auth.swift

```diff
@@ -76,91 +71,64 @@ extension OneginiModuleSwift {
             error != nil ? callback(SdkError.convertToFlutter(error)) : callback(String.stringify(json: result ?? []))
         })
     }
-    
-    func authenticateWithRegisteredAuthentication(_ identifierId: String?, callback: @escaping FlutterResult) {
-        guard let _identifierId = identifierId else {
-            callback(SdkError.convertToFlutter(SdkError(.authenticatorIdIsNull)))
-            return
+
+    func authenticateWithRegisteredAuthentication(_ profileId: String, _ registeredAuthenticatorId: String, callback: @escaping FlutterResult) {
+        guard let profile: ONGUserProfile = ONGClient.sharedInstance().userClient.userProfiles().first(where: { $0.profileId == profileId }) else {
+            return callback(SdkError(.noUserProfileIsAuthenticated).flutterError())
         }
-        guard let profile: ONGUserProfile = ONGClient.sharedInstance().userClient.userProfiles().first else
-        {
-            callback(SdkError.convertToFlutter(SdkError(.userProfileDoesNotExist)))
-            return
+
+        guard let registeredAuthenticator = ONGUserClient.sharedInstance().registeredAuthenticators(forUser: profile).first(where: { $0.identifier == registeredAuthenticatorId }) else {
+            return callback(SdkError(.authenticatorNotFound).flutterError())
         }
-        
-        let registeredAuthenticator = Array(ONGUserClient.sharedInstance().registeredAuthenticators(forUser: profile)).first(where: { $0.identifier == identifierId })
-        
-        // Preferred Authenticator
-        bridgeConnector.toAuthenticatorsHandler.setPreferredAuthenticator(profile, _identifierId) { [weak self] (value, error) in
-            
-            guard error == nil else { callback(SdkError.convertToFlutter(error))
-                return
+
+        bridgeConnector.toLoginHandler.authenticateUser(profile, authenticator: registeredAuthenticator, completion: {
+            (userProfile, error) -> Void in
+            if let _userProfile = userProfile {
```

Comment:
```
guard let userProfile = userProfile else { 
callback(SdkError.convertToFlutter(error)) // aka completion insatead of callback
}

Reply:
I still see `if let`

Reply:
whoops, fixed, also on the remaining other places